### PR TITLE
Fix missing escaping in mermaid diagram

### LINKED ISSUE #27: Mermaid diagram missing some escaping?
I tried using a NamedTuple as an input to a component and got:
```julia
  subgraph UNPACK_MODEL_InputSchema[Input: NamedTuple{(:composite_model,), Tuple{CompositeModelV1}}]
```
in my generated mermaid, which rendered as:
Screenshot 2023-09-18 at 5 36 54 PM

### PR DESCRIPTION
**Pull request by hannahilea**
Fix #27.

...sure is painful to have to make a breaking release for each update of the mermaid diagram generation...

### CODE REVIEW

**Comment by ericphanson** [comment]
Maybe we should try a really nasty type, like
```julia
julia> struct Output
          x::Val{Symbol("""\"hi1232}{{}:y;./[]]""")}
       end

julia> fieldtype(Output, 1)
Val{Symbol("\"hi1232}{{}:y;./[]]")}
```
I wonder if we would need something like `escape_string` to properly escape everything

**Comment by hannahilea** [comment]
good idea---i'll add that as a test

**Comment by hannahilea** [comment]
lol, your `Output` struct breaks my vscode linting :)

**Comment by hannahilea** [comment]
So mermaid chokes if you url-escape things, but it can otherwise handle special characters just fine as long as `"` is encoded as `"#quot;"`---so I did that.

**Comment by hannahilea** [comment]
(ready for review)

File: test/mermaid.jl [hunk 1]
```diff
@@ -24,6 +24,7 @@ end # module
 @version SchemaRadV1 begin
     foo::Union{String,Missing}
     list::Vector{Int}
+    d::Val{Symbol("""\"hi1232}{{}:yj,;./[]]""")}
```
Comment (ericphanson) [inline test/mermaid.jl:27]: hah, this also ruins github's syntax highlighting 😂

**Review by ericphanson — APPROVED** [review]
lgtm!

File: test/mermaid.jl (hunk 1 above)
Comment (ericphanson) [reply test/mermaid.jl:27]: I think this is https://github.com/JuliaEditorSupport/atom-language-julia/issues/219 (which appear to be used by GitHub via Linguist and VSCode)

**Pull request merged**